http/basic/src/main/java/io/cloudevents/http/HttpMessageFactory.java

```diff
@@ -0,0 +1,118 @@
+        return MessageUtils.parseStructuredOrBinaryMessage(
+            () -> headers.get("Content-Type"),
+            format -> new GenericStructuredMessageReader(format, body),
+            () -> headers.get(CloudEventsHeaders.SPEC_VERSION),
+            sv -> new HttpMessageReader(sv, headers.entrySet(), body),
+            UnknownEncodingMessageReader::new
+        );
+    }
+
+    public static MessageWriter createWriter(BiConsumer<String, String> putHeader, Consumer<byte[]> putBody) {
+        return new HttpMessageWriter(putHeader, putBody);
+    }
+
+    public static CloudEvent readCloudEvent(Map<String,String> headers, byte[] body) {
+        return createReader(headers, body).toEvent();
+    }
+
+    public static CloudEvent readCloudEvent(Wire wire) {
+        return createReader(wire.getHeaders(), wire.getBody()).toEvent();
+    }
+
+    public static void writeCloudEventBinary(CloudEvent cloudEvent, BiConsumer<String, String> putHeader, Consumer<byte[]> putBody) {
+        MessageWriter writer = createWriter(putHeader, putBody);
+        writer.writeBinary(cloudEvent);
+    }
+
+    public static void writeCloudEventStructured(CloudEvent cloudEvent, EventFormat eventFormat, BiConsumer<String, String> putHeader, Consumer<byte[]> putBody) {
+        MessageWriter writer = createWriter(putHeader, putBody);
+        writer.writeStructured(cloudEvent, eventFormat);
+    }
+
+    public static void writeCloudEventStructured(CloudEvent cloudEvent, String eventFormat, BiConsumer<String, String> putHeader, Consumer<byte[]> putBody) {
+        MessageWriter writer = createWriter(putHeader, putBody);
+        writer.writeStructured(cloudEvent, eventFormat);
+    }
+
+    public static Wire writeCloudEventBinary(CloudEvent cloudEvent) {
+        Wire result = new Wire();
+        Map<String,String> headers = new HashMap<>(4);
+        result.setHeaders(headers);
+        MessageWriter writer = createWriter(headers::put, bytes -> result.setBody(bytes));
+        writer.writeBinary(cloudEvent);
+        return result;
+    }
+
+    public static Wire writeCloudEventStructured(CloudEvent cloudEvent, EventFormat eventFormat) {
+        Wire result = new Wire();
+        Map<String,String> headers = new HashMap<>(4);
+        result.setHeaders(headers);
+        MessageWriter writer = createWriter(headers::put, bytes -> result.setBody(bytes));
+        writer.writeStructured(cloudEvent, eventFormat);
+        return result;
+    }
+
+    public static Wire writeCloudEventStructured(CloudEvent cloudEvent, String eventFormat) {
+        Wire result = new Wire();
+        Map<String,String> headers = new HashMap<>(4);
+        result.setHeaders(headers);
+        MessageWriter writer = createWriter(headers::put, bytes -> result.setBody(bytes));
+        writer.writeStructured(cloudEvent, eventFormat);
+        return result;
+    }
+
+    public static final class Wire {
```

Comment:
Do we want this `Wire` stuff?

Reply:
Solve the previous comment about read/write CloudEvent, then we'll find out if we still need this